CMakeLists.txt

```diff
@@ -69,143 +70,26 @@ execute_process(COMMAND patch -N -s -V never
                 ${CMAKE_CURRENT_SOURCE_DIR}/utils/variant/include/mpark/variant.hpp
                 ${CMAKE_CURRENT_SOURCE_DIR}/utils/cuda_compatibility.patch)
 
-add_library (singularity-opac::flags INTERFACE IMPORTED GLOBAL)
-add_library (singularity-opac::libs INTERFACE IMPORTED)
-
-# If the user doesn't specify a build type, prefer RelWithDebInfo
-set(default_build_type "RelWithDebInfo")
-if(NOT CMAKE_BUILD_TYPE AND NOT CMAKE_CONFIGURATION_TYPES)
-  message(STATUS "Setting build type to '${default_build_type}' as none was specified.")
-  set(CMAKE_BUILD_TYPE "${default_build_type}" CACHE
-    STRING "Choose the type of build." FORCE)
-  # Set the possible values of build type for cmake-gui
-  set_property(CACHE CMAKE_BUILD_TYPE PROPERTY STRINGS
-    "Debug" "Release" "MinSizeRel" "RelWithDebInfo")
-endif()
+# Flags and libraries for building, see cmake/SetupFlags.cmake
+#add_library (singularity-opac::flags INTERFACE IMPORTED)
```

Comment:
I got burned by combining things into just flags before for in-line builds. But I agree this is better if it works. Let's just try it but keep this as a solution in the back of our minds if we run into issues.

Reply:
See 744ae3b